compiler/resolution/functionResolution.cpp

```diff
@@ -4663,6 +4670,37 @@ resolveNew(CallExpr* call)
     call->baseExpr = arg;
     parent_insert_help(call, call->baseExpr);
 
+    if (initCall) {
+      // 4: insert the allocation for the instance and pass that in as an
+      // argument if we're making a call to an initializer
+      Type* typeToNew = toReplace->var->typeInfo();
+      VarSymbol* new_temp = newTemp(typeToNew);
```

Comment:
I think it is nice to people debugging with the C code to name these temporaries. E.g. newTemp("new_temp", typeToNew).